contribs/ev/src/main/java/org/matsim/contrib/ev/charging/FastThenSlowCharging.java

```diff
@@ -54,6 +54,47 @@ public double calcChargingPower(double maxPower) {
 		}
 	}
 
+	@Override
+	public double calcEnergyCharge(ChargerSpecification charger, double chargePeriod) {
```

Comment:
This class does a bit more than a simple mathematical operation, so please add some tests for this methods (there are some tests for the other method, maybe you can base on them).

Reply:
You are right! I added a test (and indeed there were som ebugs ;) ) thanks! fixed in 672afbf